ext/cuda/bilinearinterpolation.jl

```diff
@@ -0,0 +1,169 @@
+import ..ClimaInterpolations.BilinearInterpolation:
```

Comment:
I have similar comments for this file as for src/BilinearInterpolation.jl - could you add docstrings for all the functions, explain the interpolation algorithm being used, and comment on some of the CUDA-specific things being done (e.g. why are we able to use `nyloops` in the for loop)?